File: dynamicprogramming/Memorization/memorization.js
function addto80(n){
    return n+80
}
function memorize(n){
    let cache={}
    return (function(n){
        if(n in cache){
            return cache[n]
        }
        else{
            console.log("it takes")
            cache[n]=n+80
            return cache[n]
        }
    })
}
let mem=memorize()
console.log(mem(5))
console.log(mem(5))

File: dynamicprogramming/Memorization/memorization.py
#standard function

#dynamic programming:memorization
def memorize(n):
    cache={}
    def innermemorize(n):
        if (n in cache):
            return cache[n]
        else:
            print('long time')
            cache[n]=n+80
            return cache[n]
    return innermemorize
@memorize
def addto80(n):
    return n+80

print(addto80(5))
print(addto80(5))
